assets/js/modules/analytics-4/components/common/WebDataStreamSelect.js

```diff
@@ -0,0 +1,221 @@
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAccountID()
+	);
+
+	const { propertyID, webDataStreamID, measurementID } = useSelect(
+		( select ) => select( MODULES_ANALYTICS_4 ).getSettings() || {}
+	);
+
+	const webDataStreams = useSelect( ( select ) =>
+		isValidPropertyID( propertyID )
+			? select(
+					MODULES_ANALYTICS_4
+			  ).getMatchingWebDataStreamsByPropertyID( propertyID )
+			: []
+	);
+
+	const isLoading = useSelect( ( select ) => {
+		if ( isDisabled ) {
+			return false;
+		}
+
+		const loadedAccounts =
+			select( MODULES_ANALYTICS ).hasFinishedResolution( 'getAccounts' );
+
+		const loadedProperties = select(
+			MODULES_ANALYTICS_4
+		).hasFinishedResolution( 'getProperties', [ accountID ] );
+
+		const loadedWebDataStreams = isValidPropertyID( propertyID )
+			? select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+					'getWebDataStreams',
+					[ propertyID ]
+			  )
+			: true;
+
+		const finishedSelectingAccount =
+			select( MODULES_ANALYTICS ).hasFinishedSelectingAccount() !== false;
+
+		const isMatchingAccountProperty =
+			select( MODULES_ANALYTICS_4 ).isMatchingAccountProperty();
+
+		return (
+			isMatchingAccountProperty ||
+			! loadedAccounts ||
+			! loadedProperties ||
+			! loadedWebDataStreams ||
+			! finishedSelectingAccount
+		);
+	} );
+
+	const viewContext = useViewContext();
+	const { setWebDataStreamID, updateSettingsForMeasurementID } =
+		useDispatch( MODULES_ANALYTICS_4 );
+	const onWebDataStreamChange = useCallback(
+		( index, { dataset } ) => {
+			const newID = dataset.value;
+			if ( webDataStreamID === newID ) {
+				return;
+			}
+
+			setWebDataStreamID( newID );
+			updateSettingsForMeasurementID(
+				webDataStreams.find( ( { _id } ) => _id === newID )
+					?.webStreamData?.measurementId || '' // eslint-disable-line sitekit/acronym-case
+			);
+
+			trackEvent(
+				`${ viewContext }_analytics`,
+				newID === WEBDATASTREAM_CREATE
+					? 'change_webdatastream_new'
+					: 'change_webdatastream',
+				'ga4'
+			);
+		},
+		[
+			webDataStreams,
+			webDataStreamID,
+			setWebDataStreamID,
+			updateSettingsForMeasurementID,
+			viewContext,
+		]
+	);
+
+	if ( ! isValidAccountID( accountID ) ) {
+		return null;
+	} else if ( isLoading ) {
+		return <ProgressBar height={ 100 } small />;
+	}
+
+	const isValidSelection =
+		webDataStreamID === undefined || webDataStreamID === ''
+			? true
+			: isValidWebDataStreamSelection( webDataStreamID );
+
+	if ( hasModuleAccess === false ) {
+		return (
+			<Select
+				className={ classnames(
+					'googlesitekit-analytics-4__select-measurement',
+					className
+				) }
+				label={ __( 'Web Data Stream', 'google-site-kit' ) }
+				value={ measurementID }
+				enhanced
+				outlined
+				disabled
+			>
+				<Option value={ measurementID }>{ measurementID }</Option>
+			</Select>
+		);
+	}
+
+	return (
+		<Select
+			className={ classnames(
+				'googlesitekit-analytics-4__select-measurement',
```

Comment:
Let's update the class to be consistent with the updating naming convention as well.

Reply:
Yep, updated.